osu.Game/Skinning/Editor/SkinSelectionHandler.cs

```diff
@@ -234,18 +250,82 @@ private void applyOrigin(Anchor anchor)
         private Quad getSelectionQuad() =>
             GetSurroundingQuad(SelectedBlueprints.SelectMany(b => b.Item.ScreenSpaceDrawQuad.GetVertices().ToArray()));
 
-        private void applyAnchor(Anchor anchor)
+        private void applyFixedAnchors(Anchor anchor)
         {
             foreach (var item in SelectedItems)
             {
                 var drawable = (Drawable)item;
 
-                var previousAnchor = drawable.AnchorPosition;
-                drawable.Anchor = anchor;
-                drawable.Position -= drawable.AnchorPosition - previousAnchor;
+                item.UsesFixedAnchor = true;
+                applyAnchor(drawable, anchor);
             }
         }
 
+        private void applyClosestAnchors()
+        {
+            foreach (var item in SelectedItems)
+            {
+                item.UsesFixedAnchor = false;
+                applyClosestAnchor((Drawable)item);
+            }
+        }
+
+        private static Anchor getClosestAnchor(Drawable drawable)
+        {
+            var parent = drawable.Parent;
+
+            if (parent == null)
+                return drawable.Anchor;
+
+            var screenPosition = getOriginPositionFromQuad(drawable.ScreenSpaceDrawQuad, drawable.Origin);
+            var absolutePosition = parent.ToLocalSpace(screenPosition);
+            var factor = parent.RelativeToAbsoluteFactor;
+
+            var result = default(Anchor);
+
+            result |= getTieredComponent(absolutePosition.X / factor.X, Anchor.x0, Anchor.x1, Anchor.x2);
+            result |= getTieredComponent(absolutePosition.Y / factor.Y, Anchor.y0, Anchor.y1, Anchor.y2);
+
+            return result;
+        }
+
+        private static Vector2 getOriginPositionFromQuad(in Quad quad, Anchor origin)
+        {
+            var result = quad.TopLeft;
+
+            if (origin.HasFlagFast(Anchor.x2))
+                result.X += quad.Width;
+            else if (origin.HasFlagFast(Anchor.x1))
+                result.X += quad.Width / 2f;
+
+            if (origin.HasFlagFast(Anchor.y2))
+                result.Y += quad.Height;
+            else if (origin.HasFlagFast(Anchor.y1))
+                result.Y += quad.Height / 2f;
+
+            return result;
+        }
+
+        private static Anchor getTieredComponent(float component, Anchor tier0, Anchor tier1, Anchor tier2)
+        {
+            if (component >= 2 / 3f)
+                return tier2;
+
+            if (component >= 1 / 3f)
+                return tier1;
+
+            return tier0;
+        }
```

Comment:
Should this just be moved to `DrawableExtensions` (and `getClosestAnchor` be made `public`)? Nothing else is using it, but it would reduce some bloat of `SkinSelectionHandler`, and it is a pure function.

Reply:
make it a local function inside the only usage site. also don't use the word "tier". probably just `anchor0` .. `anchor2`.